packages/test/test-end-to-end-tests/src/test/data-virtualization/groupIdInSummary.spec.ts

```diff
@@ -82,21 +90,67 @@ describeCompat("Create data store with group id", "NoCompat", (getTestObjectProv
 		groupId: string | undefined,
 		message: string,
 	) => {
-		assert(snapshotTree.omitted, message);
 		assert(snapshotTree.groupId === groupId, message);
 		assert(Object.entries(snapshotTree.trees).length === 0, message);
 		assert(Object.entries(snapshotTree.blobs).length === 0, message);
 	};
 
+	const assertOmittedBlobContents = (
+		snapshot: ISnapshotTree | ISnapshot,
+		groupId: string | undefined,
+		blobContents: Map<string, ArrayBuffer>,
+		message: string,
+	) => {
+		const snapshotTree = getSnapshotTree(snapshot);
+		assert(snapshotTree.groupId === groupId, message);
+		assert(assertOmittedBlobContentsCore(snapshotTree, groupId, blobContents), message);
+	};
+
+	const assertOmittedBlobContentsCore = (
+		snapshotTree: ISnapshotTree,
+		groupId: string | undefined,
+		blobContents: Map<string, ArrayBuffer>,
+	): boolean | undefined => {
+		for (const id of Object.values(snapshotTree.blobs)) {
+			// Even if 1 blob is missing, return true.
+			if (!blobContents.has(id)) {
+				return true;
+			}
+		}
+		for (const tree of Object.values(snapshotTree.trees)) {
+			if (tree.groupId === undefined) {
+				const omitted = assertOmittedBlobContentsCore(tree, groupId, blobContents);
+				if (omitted) {
+					return true;
+				}
+			}
+		}
+	};
+
 	const assertPopulatedTree = (
 		snapshotTree: ISnapshotTree,
 		groupId: string | undefined,
+		blobContents: Map<string, ArrayBuffer>,
 		message: string,
 	) => {
-		assert(snapshotTree.omitted === undefined, message);
 		assert(snapshotTree.groupId === groupId, message);
-		assert(Object.entries(snapshotTree.trees).length > 0, message);
-		assert(Object.entries(snapshotTree.blobs).length > 0, message);
+		assertPopulatedTreeCore(snapshotTree, groupId, blobContents, message);
```

Comment:
This could be one function, but that's more of a nit, feel free to resolve and ignore.